internal/controller/standalone_pgadmin/users.go

```diff
@@ -0,0 +1,286 @@
+		err := json.Unmarshal(existingUserSecret.Data["users.json"], &existingUsersArr)
+		if err != nil {
+			return err
+		}
+	}
+	existingUsersMap := make(map[string]secretPGAdminUser)
+	for _, user := range existingUsersArr {
+		existingUsersMap[user.Username] = user
+	}
+	intentUsers := []secretPGAdminUser{}
+	for _, user := range pgadmin.Spec.Users {
+		var stdin, stdout, stderr bytes.Buffer
+		typeFlag := "--nonadmin"
+		if user.IsAdmin {
+			typeFlag = "--admin"
+		}
+
+		// Assemble user that will be used in add/update command and in updating
+		// the users.json file in the secret
+		intentUser := secretPGAdminUser{
+			Username: user.Username,
+			Password: "",
+			IsAdmin:  user.IsAdmin,
+		}
+		// If the user already exists in users.json, and isAdmin has changed, run
+		// the update-user command. If the user already exists in users.json, but
+		// it hasn't changed, do nothing. If the user doesn't exist in users.json,
+		// run the add-user command.
+		if existingUser, present := existingUsersMap[user.Username]; present {
+			// Set password for intentUser
+			intentUser.Password = existingUser.Password
+
+			if user.IsAdmin != existingUser.IsAdmin {
+				// Attempt update-user command
+				script := setupScript + fmt.Sprintf(`python3 setup.py update-user %s "%s"`,
+					typeFlag, intentUser.Username) + "\n"
+				err = exec(ctx, &stdin, &stdout, &stderr,
+					[]string{"bash", "-ceu", "--", script}...)
+
+				// If any errors occurred during update, we want to log a message,
+				// add the existing user to users.json since the update was
+				// unsuccessful, and continue reconciling users.
+				if err != nil {
+					log.Error(err, "PodExec failed: ")
+					intentUsers = append(intentUsers, existingUser)
+					continue
+				} else if strings.TrimSpace(stderr.String()) != "" {
+					log.Error(errors.New(stderr.String()), fmt.Sprintf("pgAdmin setup.py error for %s: ",
+						intentUser.Username))
+					intentUsers = append(intentUsers, existingUser)
+					continue
+				}
+			}
+		} else {
+			// New user, so generate a password and set it on intentUser
+			password, err := util.GenerateASCIIPassword(util.DefaultGeneratedPasswordLength)
+			if err != nil {
+				return err
+			}
+			intentUser.Password = password
+
+			// Attempt add-user command
+			script := setupScript + fmt.Sprintf(`python3 setup.py add-user %s -- "%s" "%s"`,
+				typeFlag, intentUser.Username, intentUser.Password) + "\n"
+			err = exec(ctx, &stdin, &stdout, &stderr,
+				[]string{"bash", "-ceu", "--", script}...)
+
+			// If any errors occurred when attempting to add user, we want to log a message,
+			// and continue reconciling users.
+			if err != nil {
+				log.Error(err, "PodExec failed: ")
+				continue
+			}
+			if strings.TrimSpace(stderr.String()) != "" {
+				log.Error(errors.New(stderr.String()), fmt.Sprintf("pgAdmin setup.py error for %s: ",
+					intentUser.Username))
+				continue
+			}
+			if strings.Contains(stdout.String(), "email address") ||
```

Comment:
:thinking: If I understand correctly, these two `Contains` checks are looking for particular pgAdmin errors. Is that right? 
If so, could you link to the errors in question in the pgAdmin code? That may help down the road if the messages change.